service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/SendDocumentControllerTest.java

```diff
@@ -0,0 +1,134 @@
+import org.apache.commons.lang3.RandomUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.enums.RepresentativeRole;
+import uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences;
+import uk.gov.hmcts.reform.fpl.model.DocumentsSentToParty;
+import uk.gov.hmcts.reform.fpl.model.Representative;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.DocumentDownloadService;
+import uk.gov.hmcts.reform.fpl.utils.ElementUtils;
+import uk.gov.hmcts.reform.sendletter.api.LetterWithPdfsRequest;
+import uk.gov.hmcts.reform.sendletter.api.SendLetterApi;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+
+import static java.time.ZoneOffset.UTC;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences.DIGITAL_SERVICE;
+import static uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences.EMAIL;
+import static uk.gov.hmcts.reform.fpl.enums.RepresentativeServingPreferences.POST;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+import static uk.gov.hmcts.reform.fpl.utils.TestDataHelper.testDocument;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(SendDocumentController.class)
+@OverrideAutoConfiguration(enabled = true)
+class SendDocumentControllerTest extends AbstractControllerTest {
+
+    private static final byte[] PDF = {1, 2, 3, 4, 5};
+    private final String serviceAuthToken = RandomStringUtils.randomAlphanumeric(10);
+
+    @MockBean
+    private Clock clock;
+
+    @MockBean
+    private DocumentDownloadService documentDownloadService;
+
+    @MockBean
+    private LDClient ldClient;
+
+    @MockBean
+    private AuthTokenGenerator authTokenGenerator;
+
+    @MockBean
+    private SendLetterApi sendLetterApi;
+
+    SendDocumentControllerTest() {
+        super("send-document");
+    }
+
+    @BeforeEach
+    void setupStoppedClock() {
+        given(clock.getZone()).willReturn(UTC.normalized());
+        given(clock.instant()).willReturn(LocalDateTime.of(2020, 1, 5, 12, 10).toInstant(UTC));
```

Comment:
Do you need a clock specifically, we already have a FixedTimeConfig in the unit tests that can get a fixed `LocalDateTime`

Reply:
fixed